src/handler/mod.rs

```diff
@@ -0,0 +1,141 @@
+/// This module most of low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Result};
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<()> {
+    if !force {
+        match get_boot_counter() {
+            Some(t) if t <= 0 => bail!("countdown ended, check greenboot-rollback status"),
+            None => bail!("boot_counter is not set"),
+            _ => {
+                log::info!("restarting system");
+                let status = Command::new("systemctl").arg("reboot").status()?;
+                if status.success() {
+                    return Ok(());
+                }
+            }
```

Comment:
if  `get_bott_counter()` is going to return an Option, then:
```suggestion
        match get_boot_counter() {
            Some(t) => {
                if t <= 0 {
                    bail!("countdown ended, check greenboot-rollback status")
                } else {
                    log::info!("restarting system");
                    let status = Command::new("systemctl").arg("reboot").status()?;
                    if status.success() {
                        return Ok(());
                    }
                }
            }
            None => bail!("boot_counter is not set"),
        }
```

Reply:
I can think of  removing `else{}` , else have to duplicate the else part outside force .